src/lib.rs

```diff
@@ -361,845 +92,302 @@ impl Navigator {
-    ///     nav.set_pwm_enable(!previous);
-    ///
-    ///     println!("Enable pin logic value is {}.", nav.get_pwm_enable());
-    ///
-    ///     sleep(Duration::from_millis(1000));
-    /// }
-    /// ```
-    pub fn get_pwm_enable(&mut self) -> bool {
-        self.pwm.oe_pin.get_value().expect("Error: Get PWM value") == 1
+    fn get_temperature_sensor(&mut self) -> Option<&mut dyn TemperatureSensor> {
+        for device in &mut self.devices {
+            if let Some(temp_sensor) = device.as_temperature_sensor() {
+                return Some(temp_sensor);
+            }
+        }
+        None
     }
 
-    /// Sets the Duty Cycle (high value time) of selected channel.
-    ///
-    /// On PCA9685, this function sets the `OFF` counter and uses ON value as 0.
-    ///
-    /// # Further info
-    /// Check **[7.3.3 LED output and PWM control](https://www.nxp.com/docs/en/data-sheet/PCA9685.pdf#page=16)**
-    ///
-    /// # Examples
-    ///
-    /// ```no_run
-    /// use navigator_rs::{Navigator, PwmChannel};
-    ///
-    /// let mut nav = Navigator::new();
-    ///
-    /// nav.init();
-    /// nav.set_pwm_enable(true);
-    ///
-    /// nav.set_pwm_freq_prescale(99); // sets the pwm frequency to 60 Hz
-    /// nav.set_pwm_channel_value(PwmChannel::Ch1, 2048); // sets the duty cycle to 50%
-    /// ```
-    pub fn set_pwm_channel_value(&mut self, channel: PwmChannel, mut value: u16) {
-        let max_value = 4095;
-        if value > max_value {
-            warn!("Invalid value. Value must be less than or equal {max_value}.");
-            value = max_value;
+    fn get_barometer_sensor(&mut self) -> Option<&mut dyn BarometerSensor> {
+        for device in &mut self.devices {
+            if let Some(baro_sensor) = device.as_barometer_sensor() {
+                return Some(baro_sensor);
+            }
         }
-        self.pwm
-            .set_channel_on_off(channel.into(), 0, value)
-            .unwrap();
+        None
     }
 
-    /// Calculate and set the necessary values for the desired Duty Cycle (high value time) of selected channel.
-    ///
-    /// This method also allows the relay mode using a value of 1.0.
-    ///
-    /// # Examples
-    ///
-    /// ```no_run
-    /// use navigator_rs::{Navigator, PwmChannel};
-    ///
-    /// let mut nav = Navigator::new();
-    ///
-    /// nav.init();
-    /// nav.set_pwm_enable(true);
-    ///
-    /// nav.set_pwm_freq_prescale(99); // sets the pwm frequency to 60 Hz
-    /// nav.set_pwm_channel_duty_cycle(PwmChannel::Ch1, 0.50); // sets the duty cycle to 50%
-    /// ```
-    pub fn set_pwm_channel_duty_cycle(&mut self, channel: PwmChannel, duty_cycle: f32) {
-        let duty_cycle = duty_cycle.clamp(0.0, 1.0);
-        let max_value = 4095;
-
-        if approx::relative_eq!(duty_cycle, 1.0) {
-            self.pwm.set_channel_full_on(channel.into(), 0).unwrap();
-            return;
+    fn get_magnetometer_sensor(&mut self) -> Option<&mut dyn MagnetometerSensor> {
```

Comment:
How the newer API will handle the second Magnetometer?

Reply:
It'll return a vector of an option